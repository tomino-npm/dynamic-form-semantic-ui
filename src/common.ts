import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type SignatureType = {
  comment: string;
  signature: string;
  verifiedState: 'Pending' | 'Verified' | 'Rejected';
  rejected?: boolean;
  uid: string;
  name: string;
  date: Date;
  setValue?(name: string, value: any): any;
};

export type SignatureHandlers = {
  validateForm(): boolean;
  verifySignature(uid: string, signature: string): Promise<string>;
  sign(
    source: string,
    reject: boolean,
    password: string,
    reason: string,
    submit: boolean
  ): Promise<SignatureType>;
  signatureFont(): string;
};

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: any };
  readOnly: boolean;
};

export const { css, renderCss } = initCss('form-');
