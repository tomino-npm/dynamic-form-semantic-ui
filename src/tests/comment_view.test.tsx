import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

describe('Form', () => {
  const schema: JSONSchema = {
    type: 'object',
    properties: {
      description: {
        type: 'string'
      }
    },
    required: ['name']
  };

  const formDefinition: FormDefinition = create.form({
    elements: [
      create.formElement({
        row: 0,
        column: 0,
        width: 16,
        control: 'Comment',
        label: 'Description',
        source: 'description'
      })
    ]
  });

  const formData = {
    description: 'Some Description'
  };

  describe('Comment', () => {
    function componentWithData() {
      const form = new FormModel(formDefinition, schema, formData);

      // just another notation
      return <TestComponent form={form} />;
    }

    it('renders correctly', () => {
      const component = renderer.create(componentWithData());
      expect(component).toMatchSnapshot();
    });

    return { componentWithData };
  });
});
