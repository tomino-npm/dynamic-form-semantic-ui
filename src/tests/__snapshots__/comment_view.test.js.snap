// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <div
    className="ui dividing header"
  >
    Standard
  </div>
  <style>
    
  </style>
  <div
    className="ui form "
  >
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <div
          className="ui yellow icon message"
        >
          <i
            aria-hidden="true"
            className="comments outline icon"
            onClick={[Function]}
          />
          <div
            className="content"
          >
            <div
              className="header"
            >
              Comment
            </div>
            <p>
              Some Description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui dividing header"
  >
    Readonly
  </div>
  <style>
    
  </style>
  <div
    className="ui form form-hjcaghiia"
  >
    <div
      className="fields"
    >
      <div
        className="sixteen wide field"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <div
          className="ui yellow icon message"
        >
          <i
            aria-hidden="true"
            className="comments outline icon"
            onClick={[Function]}
          />
          <div
            className="content"
          >
            <div
              className="header"
            >
              Comment
            </div>
            <p>
              Some Description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
